climpred/utils.py

```diff
@@ -97,3 +103,52 @@ def reduce_time_series(forecast, reference, nlags):
     forecast = forecast.where(forecast.time >= imin, drop=True)
     reference = reference.where(reference.time >= imin, drop=True)
     return forecast, reference
+
+
+def assign_climpred_compute_to_attrs(
+    skill,
+    ds,
+    function_name='compute_perfect_model',
+    metadata_dict=None,
+    metric=None,
+    comparison=None,
+):
+    """Write information about prediction skill into attrs."""
+    # assign old attrs
+    skill.attrs = ds.attrs
+
+    # climpred info
+    skill.attrs[
+        'Prediction skill'
+    ]: f'calculated by climpred https://climpred.readthedocs.io/'
+    skill.attrs['skill calculated by function'] = function_name
+    if 'init' in ds.coords:
+        skill.attrs['number of initializations'] = ds.init.size
+    if 'member' in ds.coords:
+        skill.attrs['number of members'] = ds.member.size
+
+    if 'perfect_model' in function_name:
+        comparison = get_comparison_function(comparison, PM_COMPARISONS).__name__[1:]
+        metric = get_metric_function(metric, PM_METRICS).__name__[1:]
+    elif 'hindcast' in function_name:
+        comparison = get_comparison_function(comparison, HINDCAST_COMPARISONS).__name__[
+            1:
+        ]
+        metric = get_metric_function(metric, HINDCAST_METRICS).__name__[1:]
+
+    skill.attrs['metric'] = metric
+    skill.attrs['comparison'] = comparison
+
+    if metric in ['pearson_r']:
+        skill.attrs['units'] = ' '
+
+    # write optional information
+    if metadata_dict is None:
+        metadata_dict = dict()
+    for key, val in metadata_dict.items():
+        skill.attrs[key] = val
+
+    skill.attrs[
+        'date'
+    ] = f'created on {datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S%f")[:-6]}'
```

Comment:
always got an error (F722, couldnt google the root cause of the problem) with `pre-commit` when not having the f before the string. but not it doesnt show up at all. the link seems to be a problem for flake8 somehow. ideas? @ahuang11